### PR TITLE
gf180 uart-blocks test design added

### PR DESCRIPTION
**Pull request by vijayank88**
`BLOCKS` based design added for `gf180` pdk.

### CODE REVIEW

File: flow/designs/gf180/uart-blocks/uart_rx/BLOCKS_grid_strategy.tcl [hunk 1]
```diff
@@ -0,0 +1,24 @@
+####################################
```
Comment (maliberty) [inline flow/designs/gf180/uart-blocks/uart_rx/BLOCKS_grid_strategy.tcl]: Why do we have both this and flow/designs/gf180/uart-blocks/BLOCKS_grid_strategy.tcl  ?

**Review by maliberty — CHANGES_REQUESTED** [review]
Please add to the appropriate CI pipeline(s)

File: flow/designs/gf180/uart-blocks/uart_rx/BLOCKS_grid_strategy.tcl (hunk 1 above)
Comment (vijayank88) [reply flow/designs/gf180/uart-blocks/uart_rx/BLOCKS_grid_strategy.tcl]: For hardening max routing layer should be Metal4.
In the top design Metal5 used as max routing layer.
So PDN adjusted accordingly with different strategy

**Comment by vijayank88** [comment]
In asap7 also same unit test there, but not part of CI. So skipped for gf180 as well. Shall I include both?

**Comment by maliberty** [comment]
> In asap7 also same unit test there, but not part of CI. So skipped for gf180 as well. Shall I include both?

Yes.  All designs should be tested otherwise they tend to rot.

File: flow/designs/gf180/uart-blocks/BLOCKS_grid_strategy.tcl [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# voltage domains
+####################################
+set_voltage_domain -name {CORE} -power {VDD} -ground {VSS}
+####################################
+# standard cell grid
+####################################
+define_pdn_grid -name {block} -voltage_domains {CORE} 
+add_pdn_stripe -grid {block} -layer {Metal1} -width {0.900} -pitch {5.040} -offset {0} -followpins
+add_pdn_stripe -grid {block} -layer {Metal4} -width {4.480} -spacing {0.56} -pitch {44.8} -offset {22.4}
+if { [info exists ::env(BLOCKS)] && [expr {$::env(BLOCKS) == $::env(DESIGN_NAME)}] } {
```
Comment (vijayank88) [inline flow/designs/gf180/uart-blocks/BLOCKS_grid_strategy.tcl]: @vvbandeira 
In `config.mk`, variable `BLOCKS` defined. But why I'm unable to read the variable in this tcl file?

Comment (vvbandeira) [reply flow/designs/gf180/uart-blocks/BLOCKS_grid_strategy.tcl]: What makes you think the variable is not being read?

FYI, from a quick glance at the PR and its files, it does not look like the if expression will ever evaluate to true. There's only one place where you set `export BLOCKS = uart_rx` and no file where `DESIGN_NAME` is `uart_rx`.

Comment (vijayank88) [reply flow/designs/gf180/uart-blocks/BLOCKS_grid_strategy.tcl]: With `BLOCKS` variable `uart_rx` will be hardened before running `uart-blocks`.

During `uart_rx` run, I want `if` condition to run. For `uart-blocks`, `else` to run to use all metal layers.

Is possible to implement?

Comment (vijayank88) [reply flow/designs/gf180/uart-blocks/BLOCKS_grid_strategy.tcl]: removed the if else condition. Later we will check for this implementation

**Comment by vijayank88** [comment]
@maliberty 
Is the CI error related to what @oharboe reported?
```
==========================================================================

finish report_design_area

--------------------------------------------------------------------------

Design area 604 u^2 67% utilization.


Signal 11 received

Stack trace:

 0# 0x0000000000D3CBC7 in /tmp/workspace/ublic_tests_all-pr_PR-1136-merge/tools/install/OpenROAD/bin/openroad

 1# 0x00007F4F8658E400 in /lib64/libc.so.6

 2# gui::DbInstDescriptor::getInstanceType(odb::dbInst*) const in /tmp/workspace/ublic_tests_all-pr_PR-1136-merge/tools/install/OpenROAD/bin/openroad

 3# gui::DisplayControls::getInstRow(odb::dbInst*) const in /tmp/workspace/ublic_tests_all-pr_PR-1136-merge/tools/install/OpenROAD/bin/openroad

 4# non-virtual thunk to gui::DisplayControls::isInstanceVisible(odb::dbInst*) in /tmp/workspace/ublic_tests_all-pr_PR-1136-merge/tools/install/OpenROAD/bin/openroad

 5# gui::RenderThread::drawBlock(QPainter*, odb::dbBlock*, odb::Rect const&, int) in /tmp/workspace/ublic_tests_all-pr_PR-1136-merge/tools/install/OpenROAD/bin/openroad

 6# gui::RenderThread::draw(QImage&, QRect const&, std::set<gui::Selected, std::less<gui::Selected>, std::allocator<gui::Selected> > const&, std::array<std::set<gui::Selected, std::less<gui::Selected>, std::allocator<gui::Selected> >, 8ul> const&, std::vector<std::unique_ptr<gui::Ruler, std::default_delete<gui::Ruler> >, std::allocator<std::unique_ptr<gui::Ruler, std::default_delete<gui::Ruler> > > > const&, double, QColor const&) in /tmp/workspace/ublic_tests_all-pr_PR-1136-merge/tools/install/OpenROAD/bin/openroad

 7# gui::RenderThread::run() in /tmp/workspace/ublic_tests_all-pr_PR-1136-merge/tools/install/OpenROAD/bin/openroad

 8# 0x00007F4F87715E71 in /lib64/libQt5Core.so.5

 9# 0x00007F4F8B014EA5 in /lib64/libpthread.so.0

10# clone in /lib64/libc.so.6

Elapsed time: 0:01.00[h:]min:sec. CPU time: user 0.84 sys 0.08 (92%). Peak memory: 117048KB.

make: *** [Makefile:698: logs/nangate45/gcd/base/6_report.log] Error 11

+ ret=2

+ '[' -z '' ']'

+ PRIVATE_DIR=../../private_tool_scripts

+ '[' -f ../../private_tool_scripts/openRoad/private.mk ']'

+ '[' -f ../../private_tool_scripts/util/utils.mk ']'

+ set -e

+ '[' '!' -z x ']'

+ make DESIGN_CONFIG=./designs/nangate45/gcd/config.mk final_report_issue

+ tee -a ./logs/nangate45/gcd.log

[INFO][FLOW] Using platform directory ./platforms/nangate45

./util/makeIssue.sh final_report

Creating run-me-gcd-nangate45-base.sh script

Creating vars-gcd-nangate45-base.sh/tcl script

Archiving issue to final_report_gcd_nangate45_base_2023-06-16_08-25.tar.gz

+ exit 2

script returned exit code 2
```

**Comment by maliberty** [comment]
Yes, it is happening randomly.

**Comment by vijayank88** [comment]
@maliberty 
Please merge this. Regarding customizing pdn we check it later.

**Review by maliberty — APPROVED** [review]

**Pull request merged**